ed_sensor_integration/src/kinect/fitter.cpp

```diff
@@ -458,6 +454,19 @@ void Fitter::checkExpectedBeamThroughEntity(const std::vector<double>& model_ran
 
 // ----------------------------------------------------------------------------------------------------
 
+void Fitter::configureBeamModel(const image_geometry::PinholeCameraModel& caminfo)
+{
+    uint nr_beams = std::min(200, caminfo.fullResolution().width); // don't use more data points than the resolution of your camera
+    double fx = caminfo.fx();
+    double fx_resize = fx * nr_beams/caminfo.fullResolution().width; //reducint nr of data points will require a differenc focal length
+    double w = 2 * double(nr_beams) / fx_resize; // reverse calculation of the width of the beam model.
+    beam_model_.initialize(w, nr_beams);
+    nr_data_points_ = nr_beams;
+    ROS_INFO("Configured fitter with %i beams and a focal length of %f", nr_beams, fx);
+    configured_ = true;
```

Comment:
I believe this math is correct but I still need to check that the image is mapped to the beam model correctly.

Reply:
After testing I can confirm that this method will make it so that the range is no bigger than the view of the camera. One may sometime wish to use a focal length smaller than fx_resize but if it is greater you will get leading and trailing zeros in ranges